### ISSUE TITLE
Accuracy measures / summary performance of ensemble models fitted using makeStackedLearner

### ISSUE THREAD

**Issue by thengl** [mlr-org/mlr#2598]
makeStackedLearner function is very useful. I especially appreciate diversity of 'stacking' methods. I am now looking for a quick way to print out summary performance measures of fitted ensemble models:

- average accuracy,
- most important individual models / weights,

In short, something that the [SuperLearner package gives out by default](https://cran.r-project.org/web/packages/SuperLearner/vignettes/Guide-to-SuperLearner.html#fit-multiple-models).

**Comment by larskotthoff** [comment]
You can get the weights through `model$learner.model$weights`. For accuracy, you can simply use the `performance()` function unless I'm missing something?

**Comment by thengl** [comment]
What about cross-validation accuracy results per each individual model? Thanks in advance.

**Comment by larskotthoff** [comment]
You can't get that directly, but you can get the individual models for a stacked learner through `res$models[[1]]$learner.model$base.models[[1]]` for example, assuming that `res` is the result of a resample operation with `models = TRUE`. You can then evaluate the performance of each as you would do with any other mlr model.

**Comment by thengl** [comment]
That would do. Many thanks. Five ***** for your package and feedback.

**Comment by larskotthoff** [comment]
Thanks -- some of this is maybe not as easy as in other packages, but in general with a bit of digging you can find/generate all the information you need.

**Comment by thengl** [comment]
When I run the stacked learner I get the model but I still do not know where to find weights based on SuperLearner:

```
data(BostonHousing, package = "mlbench")
tsk = makeRegrTask(data = BostonHousing, target = "medv")
base = c("regr.rpart", "regr.svm")
lrns = lapply(base, makeLearner)
m = makeStackedLearner(base.learners = lrns,
predict.type = "response", method = "compress")
tmp = train(m, tsk)
tmp$learner.model$weights
```
Looks like weights are not at the place you indicated?

```
> names(tmp$learner.model)
[1] "method"        "base.learners" "super.model"   "pred.train" 
```

**Comment by larskotthoff** [comment]
Well in this particular case you're training a neural network to combine the models, so mlr doesn't know about the weights. You can however get the NN model through `tmp$learner.model$super.model$learner.model` and weights through `tmp$learner.model$super.model$learner.model$wts` -- what exactly these numbers mean depends on the implementation of `regr.nnet` and you should be able to find more information on that in its documentation.

I should clarify that in general where you can find the information (or even if it will be available) depends on your particular setup of the stacked learner.

**Comment by thengl** [comment]
I am actually interested in getting weights per model for the method "stack.cv" i.e. the SuperLearner approach:

```
data(BostonHousing, package = "mlbench")
tsk = makeRegrTask(data = BostonHousing, target = "medv")
base = c("regr.rpart", "regr.svm")
lrns = lapply(base, makeLearner)
m = makeStackedLearner(base.learners = lrns, super.learner = "regr.glmnet",
                       predict.type = "response", method = "stack.cv")
tmp = train(m, tsk)
tmp$learner.model$super.model$learner.model$wts
NULL
```

It appears that weights are missing in this case. In any case it would be of course easier if the numbers are named based on the individual learner/model e.g. "regr.rpart.J1". Thank you.

**Comment by larskotthoff** [comment]
Here again mlr doesn't have explicit knowledge of any weights as you're asking it to train a model that produces the final predictions (here a glmnet model). You can extract that model through `tmp$learner.model$super.model$learner.model` and inspect as you wish (although for this type of model there are no explicit weights for the input models).

In general, every time a model is trained for something, the information on what it does is contained in that model -- mlr doesn't extract/process/aggregated that at all, but rather gives you access to the model so that you can do all of that (and anything else you might want to do) yourself.

**Comment by thengl** [comment]
In the case of the SuperLearner, the weights are basically beta coefficients from the GLM (fitted via regr.glmnet). How to get beta coefficients? I have been using previously the subsemble package (https://github.com/ledell/subsemble) and here the betas can be extracted directly. For mlr I do not know where to look anymore. I would expect that `coef(tmp$learner.model$super.model$learner.model` could do the job but this gives a long dgCMatrix. Many thanks.

**Comment by thengl** [comment]
I think I found it. I have now used as super.leaner model `regr.glm`, so that the coefficients can be extracted via `str(m@spModel$learner.model$super.model$learner.model$coefficients)`. This seems to work fine.

**Comment by larskotthoff** [comment]
Great! For glmnet, you can get the betas through `tmp$learner.model$super.model$learner.model$beta`. When you're in the R console, you can autocomplete from e.g. `mp$learner.model$super.model$learner.model$` by pressing tab twice. This will list all members in this case.
